notifications/src/main/java/dk/shape/games/notifications/presentation/SubjectNotificationsInteractor.kt

```diff
@@ -38,41 +39,44 @@ class SubjectNotificationsInteractor(
+                            subscription?.types?.mapNotNull { subscriptionType ->
+                                notificationsGroup.notificationTypes.find { notificationType ->
+                                    notificationType.identifier.name.toLowerCase(Locale.ROOT) == subscriptionType
+                                }
+                            }?.toSet() ?: emptySet()
 
-                            val defaultIdentifiers = if (enabledNotificationTypes.isEmpty()) {
-                                notificationsGroup.defaultNotificationTypeIdentifiers.toSet()
-                            } else emptySet()
+                        val defaultIdentifiers = if (enabledNotificationTypes.isEmpty()) {
+                            notificationsGroup.defaultNotificationTypeIdentifiers.toSet()
+                        } else emptySet()
 
-                            val possibleNotificationTypes =
-                                notificationsGroup.notificationTypes.toSet()
+                        val possibleNotificationTypes =
+                            notificationsGroup.notificationTypes.toSet()
 
-                            withContext(Dispatchers.Main) {
-                                if (notificationsEventHandler is SubjectNotificationsEventHandler.Full.State) {
-                                    notificationsEventHandler.onNotificationsLoading(false)
-                                }
-                                onLoaded(
-                                    enabledNotificationTypes,
-                                    possibleNotificationTypes,
-                                    defaultIdentifiers
-                                )
+                        withContext(Dispatchers.Main) {
+                            if (notificationsEventHandler is SubjectNotificationsEventHandler.Full.State) {
+                                notificationsEventHandler.onNotificationsLoading(false)
                             }
+                            onLoaded(
+                                enabledNotificationTypes,
+                                possibleNotificationTypes,
+                                defaultIdentifiers
+                            )
                         }
-                    } else onFailure(IllegalArgumentException("Subject ${action.subjectId} is missing default notifications"))
+                    } else {
+                        val message = "Sport: ${action.sportId} is missing subject notifications"
+                        onFailure(IllegalArgumentException(message))
+                    }
                 }
             }
-        } catch (e: IOException) {
-            if (notificationsEventHandler is SubjectNotificationsEventHandler.Full.State) {
-                notificationsEventHandler.onNotificationsLoading(false)
-                notificationsEventHandler.onPreferencesLoadedError(e)
-            }
-            onFailure(e)
+        } catch (e: Exception) {
```

Comment:
```suggestion
       } catch (e: IOException) {
            onFailure(e)
        } catch (e: HttpException) {
            onFailure(e)
        }
```
we could have a handleFailure(e) function instead that calls the notificationsHandler